plugins/modules/idrac_network_attributes.py

```diff
@@ -0,0 +1,534 @@
+
+    def __get_idrac_firmware_version(self):
+        firm_version = self.idrac.invoke_request(method='GET', uri=GET_IDRAC_FIRMWARE_VER_URI)
+        return firm_version.json_data.get('FirmwareVersion', '')
+
+    def __get_resource_id(self):
+        odata = '@odata.id'
+        found = False
+        res_id_uri = None
+        res_id_input = self.module.params.get('resource_id')
+        res_id_members = get_dynamic_uri(self.idrac, self.base_uri, 'Members')
+        for each in res_id_members:
+            if res_id_input and res_id_input == each[odata].split('/')[-1]:
+                res_id_uri = each[odata]
+                found = True
+                break
+        if not found and res_id_input:
+            self.module.exit_json(failed=True, msg=INVALID_ID_MSG.format(
+                res_id_input, 'resource_id'))
+        elif res_id_input is None:
+            res_id_uri = res_id_members[0][odata]
+        return res_id_uri
+
+    def __get_registry_fw_less_than_6_more_than_3(self):
+        reg = {}
+        network_device_function_id = self.module.params.get('network_device_function_id')
+        registry = get_dynamic_uri(self.idrac, REGISTRY_URI, 'Members')
+        for each_member in registry:
+            if network_device_function_id in each_member.get('@odata.id'):
+                location = get_dynamic_uri(self.idrac, each_member.get('@odata.id'), 'Location')
+                if location:
+                    uri = location[0].get('Uri')
+                    attr = get_dynamic_uri(self.idrac, uri, 'RegistryEntries').get('Attributes', {})
+                    for each_attr in attr:
+                        reg.update({each_attr['AttributeName']: each_attr['CurrentValue']})
+                    break
+        return reg
+
+    def __validate_time(self, mtime):
+        curr_time, date_offset = get_current_time(self.idrac)
+        if not mtime.endswith(date_offset):
+            self.module.exit_json(failed=True, msg=MAINTENACE_OFFSET_DIFF_MSG.format(date_offset))
+        if mtime < curr_time:
+            self.module.exit_json(failed=True, msg=MAINTENACE_OFFSET_BEHIND_MSG)
+
+    def __get_redfish_apply_time(self, aplytm, rf_settings):
+        rf_set = {}
+        reboot_req = False
+        if rf_settings:
+            if 'Maintenance' in aplytm:
+                if aplytm not in rf_settings:
+                    self.module.exit_json(failed=True, msg=APPLY_TIME_NOT_SUPPORTED_MSG.format(aplytm))
+                else:
+                    rf_set['ApplyTime'] = aplytm
+                    m_win = self.module.params.get('maintenance_window')
+                    self.__validate_time(m_win.get('start_time'))
+                    rf_set['MaintenanceWindowStartTime'] = m_win.get('start_time')
+                    rf_set['MaintenanceWindowDurationInSeconds'] = m_win.get('duration')
+            else:  # assuming OnReset is always
+                if aplytm == "Immediate" and aplytm not in rf_settings:
```

Comment:
If Immediate is not in rf_settings then we should give error that this apply time choice is not supported. 
Why are we making it OnReset ?

Reply:
will change, working on it